### ISSUE TITLE
Error: oversized data

### ISSUE THREAD

**Issue by basilji** [ethereum/web3.js#1739]
While deploying contract in ethereum private blockchain using web3 

web3.eth.contract(JSON.parse(contractabi.toString('utf8')));
mycontract.new({
from: web3.eth.accounts[0],
data: codebin,
gas: 3141592
}

I am getting below error
Error: oversized data 
at Object.InvalidResponse (/home/user1/node02/event_test/node_modules/web3/lib/web3/errors.js:38:16)
at /home/user1/node02/event_test/node_modules/web3/lib/web3/requestmanager.js:86:36
at XMLHttpRequest.request.onreadystatechange (/home/user1/node02/event_test/node_modules/web3/lib/web3/httpprovider.js:128:7)
at XMLHttpRequestEventTarget.dispatchEvent (/home/user1/node02/event_test/node_modules/xhr2/lib/xhr2.js:64:18)
at XMLHttpRequest.setReadyState (/home/user1/node02/event_test/node_modules/xhr2/lib/xhr2.js:354:12)
at XMLHttpRequest._onHttpResponseEnd (/home/user1/node02/event_test/node_modules/xhr2/lib/xhr2.js:509:12)
at IncomingMessage.<anonymous> (/home/user1/node02/eventtest/node_modules/xhr2/lib/xhr2.js:469:24)
at emitNone (events.js:111:20)
at IncomingMessage.emit (events.js:208:7)
at endReadableNT (_stream_readable.js:1064:12)

**Comment by lanhao-tik** [comment]
same here

**Comment by nathanawmk** [comment]
Same here. Not sure how to put the maxCodeSize in it

**Comment by nivida** [comment]
Which version of web3 are you using?

**Comment by nathanawmk** [comment]
Since maxCodeSize parameter in genesis.json does not work, I had to fix this issue by increasing the transaction size to 64KB in tx_pool.go. To be specified, this is for Quorum, a fork from Ethereum for enterprise use.

DOS is less applicable for private Ethereum networks such as Quorum. 

Updated code as follows:
// Heuristic limit, reject transactions over 32KB to prevent DOS attacks

// UPDATED to 64KB to support the deployment of bigger contract due to the pressing need for sophisticated/complex contract in financial/capital markets - Nathan Aw

	if tx.Size() > 64*1024 {
		return ErrOversizedData
	}

References:

https://github.com/nathanawmk/quorum/blob/master/core/tx_pool.go

https://ethereum.stackexchange.com/questions/55691/oversized-data-during-truffle-migrate/56354#56354

https://ethereum.stackexchange.com/questions/56000/quorum-oversized-data-during-contract-creation-contract-code-size-limit/56353#56353

**Comment by basilji** [comment]
we were using web3 - 0.19.0 version

**Comment by subhodi** [comment]
@basilji 
Maximum data you can pass along with single transaction is 32KB. Out of that Apprx 1KB is used for transaction details and rest is actually available to the sender to add payload. Modularize your smart contract code and use multiple transactions to deploy large smart contracts.
Error is thrown from Geth side, look at this line of code https://github.com/ethereum/go-ethereum/blob/6a33954731658667056466bf7573ed1c397f4750/core/tx_pool.go#L570 Limit is 32KB to prevent DOS attacks

**Comment by shadab14meb346** [comment]
@nathanawmk 
I have done this `if tx.Size() > 128*1024 {
	return ErrOversizedData
}`.Even after doing this when I am deploying my smart contract to my private network it is still giving `Error:  oversized data`. I have also tried to find the deployedBytecode length in mycontract.json file. I found out that it is 71236.
Reference:
https://ethereum.stackexchange.com/questions/59694/getting-this-error-error-oversized-data/59696?noredirect=1#comment71714_59696
this is the question we are discussing.
